modules/model/src/main/scala/io/janstenpickle/trace4cats/model/SpanStatus.scala

```diff
@@ -1,81 +1,81 @@
+  case object DeadlineExceeded extends SpanStatus("DeadlineExceeded") {
     override lazy val canonicalCode: Int = 4
     override lazy val isOk: Boolean = false
   }
-  case object NotFound extends SpanStatus with Camelcase {
+  case object NotFound extends SpanStatus("NotFound") {
     override lazy val canonicalCode: Int = 5
     override lazy val isOk: Boolean = false
   }
-  case object AlreadyExists extends SpanStatus with Camelcase {
+  case object AlreadyExists extends SpanStatus("AlreadyExists") {
     override lazy val canonicalCode: Int = 6
     override lazy val isOk: Boolean = false
   }
-  case object PermissionDenied extends SpanStatus with Camelcase {
+  case object PermissionDenied extends SpanStatus("PermissionDenied") {
     override lazy val canonicalCode: Int = 7
     override lazy val isOk: Boolean = false
   }
-  case object ResourceExhausted extends SpanStatus with Camelcase {
+  case object ResourceExhausted extends SpanStatus("ResourceExhausted") {
     override lazy val canonicalCode: Int = 8
     override lazy val isOk: Boolean = false
   }
-  case object FailedPrecondition extends SpanStatus with Camelcase {
+  case object FailedPrecondition extends SpanStatus("FailedPrecondition") {
     override lazy val canonicalCode: Int = 9
     override lazy val isOk: Boolean = false
   }
-  case object Aborted extends SpanStatus with Camelcase {
+  case object Aborted extends SpanStatus("Aborted") {
     override lazy val canonicalCode: Int = 10
     override lazy val isOk: Boolean = false
   }
-  case object OutOfRange extends SpanStatus with Camelcase {
+  case object OutOfRange extends SpanStatus("OutOfRange") {
     override lazy val canonicalCode: Int = 11
     override lazy val isOk: Boolean = false
   }
-  case object Unimplemented extends SpanStatus with Camelcase {
+  case object Unimplemented extends SpanStatus("Unimplemented") {
     override lazy val canonicalCode: Int = 12
     override lazy val isOk: Boolean = false
   }
-  case class Internal(message: String) extends SpanStatus with Camelcase {
+  case class Internal(message: String) extends SpanStatus("Internal") {
     override lazy val canonicalCode: Int = 13
     override lazy val isOk: Boolean = false
   }
-  case object Unavailable extends SpanStatus with Camelcase {
+  case object Unavailable extends SpanStatus("Unavailable") {
     override lazy val canonicalCode: Int = 14
     override lazy val isOk: Boolean = false
   }
-  case object DataLoss extends SpanStatus with Camelcase {
+  case object DataLoss extends SpanStatus("DataLoss") {
     override lazy val canonicalCode: Int = 15
     override lazy val isOk: Boolean = false
   }
-  case object Unauthenticated extends SpanStatus with Camelcase {
+  case object Unauthenticated extends SpanStatus("Unauthenticated") {
     override lazy val canonicalCode: Int = 16
     override lazy val isOk: Boolean = false
   }
+
+  implicit val eq: Eq[SpanStatus] = Eq.fromUniversalEquals
```

Comment:
Just a quick note on this; now we're not using enumeratum I think it might be better to check equality on `canonicalCode`

Reply:
Btw, are 2 internal statuses with different messages still equal?

Reply:
What should `Internal("msg").show` produce?

Reply:
see a suggestion below